src/variants/lambdatreasury/storage.mligo

```diff
@@ -0,0 +1,185 @@
+// SPDX-FileCopyrightText: 2022 Tezos Commons
+// SPDX-License-Identifier: LicenseRef-MIT-TC
+
+#if !VARIANT_STORAGE
+#define VARIANT_STORAGE
+
+#include "variants/lambda/types.mligo"
+#include "error_codes.mligo"
+
+// Xtz transfer amount cannot be 0
+let zero_mutez_err_msg = "ZERO_MUTEZ"
+
+// Xtz transfer amount cannot be smaller than `min_xtz_amount`
+let too_small_xtz_err_msg = "LOW_XTZ"
+
+// Xtz transfer amount cannot be bigger than `max_xtz_amount`
+let too_large_xtz_err_msg = "HIGH_XTZ"
+
+let fail_proposal_check_ (msg : string) : unit =
+  ([%Michelson ({| { FAILWITH } |} : (nat * string) -> unit)]
+    (fail_proposal_check, msg) : unit)
+
+type config_proposal =
+  { frozen_scale_value : nat option
+  ; frozen_extra_value : nat option
+  ; slash_scale_value : nat option
+  ; slash_division_value : nat option
+  ; max_proposal_size : nat option
+  }
+
+type xtz_transfer =
+  [@layout:comb]
+  { amount : tez
+  ; recipient : address
+  }
+
+type token_id = nat
+// FA2 transfer types
+type transfer_destination =
+  [@layout:comb]
+  { to_ : address
+  ; token_id : token_id
+  ; amount : nat
+  }
+type transfer_item =
+  [@layout:comb]
+  { from_ : address
+  ; txs : transfer_destination list
+  }
+
+type token_transfer =
+  [@layout:comb]
+  { contract_address : address
+  ; transfer_list : transfer_item list
+  }
+
+type transfer_type =
+  [@layout:comb]
+  | Xtz_transfer_type of xtz_transfer
+  | Token_transfer_type of token_transfer
+
+type transfer_proposal =
+  { agora_post_id : nat
+  ; transfers : transfer_type list
+  }
+
+// This type should define the type of the contract-extra field in storage
+// for this variant.
+type contract_extra =
+  { handler_storage : handler_storage
+  ; lambdas : lambdas
+  }
+
+type transfer_params = transfer_item list
+
+let handle_transfer (ops, transfer_type : (operation list) * transfer_type) : (operation list) =
+  match transfer_type with
+  | Token_transfer_type tt ->
+    begin
+      match (Tezos.get_entrypoint_opt "%transfer" tt.contract_address
+          : transfer_params contract option) with
+      | Some contract ->
+          (Tezos.transaction tt.transfer_list 0mutez contract) :: ops
+      | None -> (failwith fail_decision_callback : operation list)
+    end
+  | Xtz_transfer_type xt ->
+    begin
+      match (Tezos.get_contract_opt xt.recipient : unit contract option) with
+      | Some contract ->
+          (Tezos.transaction unit xt.amount contract) :: ops
+      | None -> (failwith fail_decision_callback : operation list)
+    end
```

Comment:
nitpick, but AFAIK when all that's contained in a `match` is another `match` then `begin`/`end` are not necessary:
```suggestion    match (Tezos.get_entrypoint_opt "%transfer" tt.contract_address
        : transfer_params contract option) with
    | Some contract ->
        (Tezos.transaction tt.transfer_list 0mutez contract) :: ops
    | None -> (failwith fail_decision_callback : operation list)
  | Xtz_transfer_type xt ->
    match (Tezos.get_contract_opt xt.recipient : unit contract option) with
    | Some contract ->
        (Tezos.transaction unit xt.amount contract) :: ops
    | None -> (failwith fail_decision_callback : operation list)
```

I think it looks a bit more clean, but since it's a matter of taste I'll let you decide.

P.s. if you go with it there are also other places where this can be applied.

Reply:
>when all that's contained in a match is another match then begin/end are not necessary:

This does not appear to be true, and I get a parse error without `begin..end` wrapping.